packages/backend/src/open_payments/auth/middleware.ts

```diff
@@ -46,13 +54,24 @@ export function createTokenIntrospectionMiddleware({
       const tokenIntrospectionClient = await ctx.container.use(
         'tokenIntrospectionClient'
       )
-      const tokenInfo = await tokenIntrospectionClient.introspect(token)
-      if (!tokenInfo) {
+      let tokenInfo: TokenInfo
+      try {
+        tokenInfo = await tokenIntrospectionClient.introspect(token)
+      } catch (err) {
+        ctx.throw(401, 'Invalid Token')
+      }
+      if (!tokenInfo.active) {
         ctx.throw(401, 'Invalid Token')
       }
+      ctx.clientKey = tokenInfo.key.jwk
+
+      if (!tokenInfo.access) {
+        await next()
+        return
+      }
       // TODO
       // https://github.com/interledger/rafiki/issues/835
-      const access = tokenInfo.access.find((access) => {
+      const access = tokenInfo.access.find((access: Access) => {
```

Comment:
https://github.com/interledger/rafiki/pull/921#discussion_r1069648783

Reply:
`isActiveTokenInfo` requires access to be there for payments, correct? So if there was no access, 
```
      if (!isActiveTokenInfo(tokenInfo)) {
        ctx.throw(401, 'Invalid Token')
      }
```
would throw. 
What I read is that you are suggesting that we throw a 403 there because we know the client but the token is inactive.

Reply:
That's ⬆️  what I did.